src/connect.c

```diff
@@ -220,6 +261,39 @@ int W32_CALL connect (int s, const struct sockaddr *servaddr, socklen_t addrlen)
 
   _sock_sig_restore();
   _sock_crit_stop();
+
+  if (rc == 0)
+  {
+#if 0
+    if ((socket->so_state & SS_PRIV) && socket->so_type == SOCK_DGRAM)
+    {
+      SOCK_DEBUGF ((", SS_PRIV"));
+
+      /* Clear any effect of previous ICMP errors etc.
+       */
+      socket->so_state &= ~(SS_CONN_REFUSED | SS_CANTSENDMORE | SS_CANTRCVMORE);
+      socket->so_error  = 0;
+
+      lport = socket->udp_sock->myport;
+      grab_localport (lport);
+    }
+#endif
+
+    if (rdata)  /* Must be SOCK_DGRAM */
+    {
+      _udp_Socket *udp = socket->udp_sock;
+
+      /* free new rx-buffer set in connect() / _UDP_open().
+       */
+      DISABLE();
+      _sock_free_rcv_buf ((sock_type*)udp);
+      udp->rx_data = rdata;      /* reuse previous data buffer */
+      ENABLE();
```

Comment:
This seems inefficient.  Could the allocation be avoided in `_UDP_open()`?

Reply:
I do not understand. Do you mean  "*Could the allocation be moved to _UDP_open()?*"

Reply:
I mean, `_UDP_open()` always allocates a new buffer, and then we immediately free it again here.  Skipping the malloc seems like an easy optimization, thought there might be some reason why you hadn't done it before.

Reply:
Is it even possible to have different address families in `local_addr` and `remote_addr` though?  That doesn't seem logical.

Reply:
> thought there might be some reason why you hadn't done it before.

Perhaps. I'm using clang-cl with `USE_ASAN=1` and `USE_UBSAN=1` more and more now (ref. [Makefile.all](https://github.com/gvanem/Watt-32/blob/master/src/makefile.all#L925)).
This ASAN feature (+this *Undefined Behavior Sanitizer*) is really good for hunting mem-leaks and such.

Reply:
True.  I wish we had asan on djgpp.